homeassistant/components/nws/weather.py

```diff
@@ -92,15 +101,27 @@ async def async_setup_entry(
     ):
         entity_registry.async_remove(entity_id)
 
-    async_add_entities([NWSWeather(entry.data, nws_data)], False)
+    platform = entity_platform.async_get_current_platform()
 
+    platform.async_register_entity_service(
+        "get_forecasts_extra",
+        {vol.Required("type"): vol.In(("hourly", "twice_daily"))},
+        "async_get_forecasts_extra_service",
+        supports_response=SupportsResponse.ONLY,
+    )
+
+    async_add_entities([NWSWeather(entry.data, nws_data)], False)
 
-if TYPE_CHECKING:
 
-    class NWSForecast(Forecast):
-        """Forecast with extra fields needed for NWS."""
+class ExtraForecast(TypedDict, total=False):
+    """Forecast extra fields from  NWS."""
 
-        detailed_description: str | None
+    # common attributes
+    datetime: Required[str]
+    is_daytime: bool | None
+    # extra attributes
+    detailed_description: str | None
+    short_description: float | None
```

Comment:
```suggestion
    short_description: str | None
```
Should be a string I guess